### PR TITLE
Replace wallet connect v1 with v2

### PR DESCRIPTION
**Pull request by peachbits**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204436463195824

### CODE REVIEW

File: src/hooks/useWalletConnect.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+/**
+ * Access Wallet Connect
+ */
+export function useWalletConnect(): WalletConnect {
+  const account = useSelector(state => state.core.account)
+  const currencyWallets = useWatch(account, 'currencyWallets')
+
+  // Utils
+  const publicAddresses = React.useMemo(async () => {
```
Comment (swansontec) [inline src/hooks/useWalletConnect.tsx:40]: The `useMemo` hook does not work with async values. If the wallet list updates quickly, you could get a big pile-up of dangling promises.

Instead, you can either use Sam's `useAsyncValue`, which is almost exactly list `useMemo` but handles async calculations, or you can just move this calculation into `getActiveSessions`, which is already `async`, and skip the caching altogether.

File: src/components/scenes/WcConnectScene.tsx [hunk 2]
```diff
@@ -27,6 +27,10 @@ import { SelectableRow } from '../themed/SelectableRow'
 
 interface Props extends EdgeSceneProps<'wcConnect'> {}
 
+export interface WcConnectSceneParams {
```
Comment (swansontec) [inline src/components/scenes/WcConnectScene.tsx:35]: The word "Scene" should not be part of these type names. We've been using `XyzScene` / `XyzParams`, not `XyzSceneParams`.

File: src/components/scenes/WcConnectionsScene.tsx [hunk 3]
```diff
@@ -78,7 +76,8 @@ export const WcConnectionsScene = (props: Props) => {
       />
     ))
       .then((result: string | undefined) => {
-        if (result) {
+        console.log(result)
```
Comment (swansontec) [inline src/components/scenes/WcConnectionsScene.tsx:79]: Extra log?

File: src/hooks/useWalletConnect.tsx (hunk 1 above)
Comment (swansontec) [reply src/hooks/useWalletConnect.tsx:40]: Yup, just skip the caching and turn this into a method that `getActiveSessions` can call. It's not used anywhere else.

File: src/components/services/WalletConnectService.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  }).then(res => {
+    walletConnectRef = res
+    console.log('WalletConnect initialized')
+    resolve(walletConnectRef)
+  })
+})
+
+export const WalletConnectService = () => {
+  useMount(async () => {
+    await walletConnectPromise
```
[the remark this answers is not in the document]
Comment (swansontec) [reply src/components/services/WalletConnectService.tsx]: OK, so the component does do something later. However, even in the later iteration, there is no error handling on the `useMount` hook. You could move this stuff into an async helper method, and then do `useMount(() => subscribeToWc().catch(error => showError(error)))`. That way you can still use the `async` keyword, but also get error handling on the whole thing.

**Review by swansontec — CHANGES_REQUESTED** [review]
Pretty nice overall.

File: src/components/scenes/WcConnectionsScene.tsx [hunk 5]
```diff
@@ -41,40 +43,10 @@ export const WcConnectionsScene = (props: Props) => {
...
-          }
-        }
-      }
-    }
-    return dAppConnections
-  }
+  useAsyncEffect(async () => {
+    const connections = await walletConnect.getActiveSessions()
+    setConnections(connections)
+  }, [walletConnect, props])
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/components/scenes/WcConnectionsScene.tsx:51]: Yes that is intended. The scene remains (I'm not sure why) when we navigate away to the connect and disconnect scenes. When those scenes call navigate to return to the connections scene it uses new props so we get a render and the list updates.

[the remark this answers is not in the document]
Comment (swansontec) [reply src/components/scenes/WcConnectionsScene.tsx:51]: That's fine, then. Can you comment this, since it's tricky and unexpected?

**Review by swansontec — APPROVED** [review]
Approved, but it will be a tricky rebase:

1. Don't forget to drop the first commit.
2. Merge conflict with the router types file.
3. Add a comment about the `props` being in the deps list, since it's unexpected but intended.

**Pull request merged**